### DISCUSSION TITLE
guides/noise-router/

### DISCUSSION THREAD

**Discussion by giscus** [misode/misode.github.io#237] [Site]
# guides/noise-router/

https://misode.github.io/guides/noise-router/

**Comment by kohy-creates** [comment]
words cannot express how much i love you for this

**Comment by BrackenStrike** [comment]
You are a beautiful person.

**Comment by alexmouf** [comment]
IT MAKES SENSE I LOVE YOU

**Comment by MrScautHD** [comment]
pls show us how we can add caves.

**Comment by crux-f** [comment]
This is incredible. Thank you.

**Comment by marios-andr** [comment]
Incredible work, can't wait for splines

**Comment by BarionLP** [comment]
## Best Tutorial ever!
## please continue!

**Reply by jaredrussell6893** [reply]
Indeed! Mojang's documentation on this is abysmal and, despite the fact that I am still confused at how this all works, this is by far the best I've found.

**Comment by BlendCreator** [comment]
Thank you so much for this.
Can´t wait for the splines👍

**Comment by Hetchell** [comment]
Pre-1.18, I could understand how the noise generator works. And bam, in 1.18, so many things in the noise generator section gets refactored and shifted around. In 1.19, I'm at a loss as to how things work now.

**Comment by AnAwesomGuy** [comment]
using the first json file, it doesnt give me a void world.
it gives me a water world that is completely river biome and lava under y -54
how can i make it generate a void world

**Comment by Lasbleic** [comment]
Thanks a lot for this tutorial!

**Comment by jexxadox** [comment]
please. where do i put this file after i create it?

**Reply by misode** [reply]
In `data/minecraft/worldgen/noise_settings/overworld.json` if you're modifying the vanilla overworld.
